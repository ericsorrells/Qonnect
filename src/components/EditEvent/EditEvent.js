import React from 'react';

const EditEvent = () => {
  return(
    <div>Edit Event Page :) !!!!</div>
  )
}

export default EditEvent;