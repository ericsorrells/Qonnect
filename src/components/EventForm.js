// ========================================================================================
import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
// ========================================================================================

class EventForm extends React.Component {
  constructor(props) {
    super(props)

    const { event } = props;
    this.state = {
      eventName:   event ? event.eventName    : '',
      date:        event ? event.date         : '',
      time:        event ? event.time         : '',
      location:    event ? event.location     : '',
      category:    event ? event.category     : '',
      imageUrl:    event ? event.imageUrl     : '',
      description: event ? event.description  : '',
      focused: ''
    }

    this.onFormSubmit        = this.onFormSubmit.bind(this);
    this.onNameChange        = this.onNameChange.bind(this);
    this.onTimeChange        = this.onTimeChange.bind(this);
    this.onLocationChange    = this.onLocationChange.bind(this);
    this.onCategoryChange    = this.onCategoryChange.bind(this);
    this.onImageUrlChange    = this.onImageUrlChange.bind(this); 
    this.onDescriptionChange = this.onDescriptionChange.bind(this);
  }

  onFormSubmit(e) {
    e.preventDefault();
    const { firstName, lastName } = this.props.profile;

    this.props.onSubmit({
      userName:    `${firstName} ${lastName}`,
      uid:         this.props.auth.uid,
      eventName:   this.state.eventName,
      date:        moment(this.state.date).valueOf(),
      time:        this.state.time,
      location:    this.state.location,
      category:    this.state.category,
      imageUrl:    this.state.imageUrl,
      description: this.state.description
    })
  }

  onNameChange(e) {
    this.setState({ eventName: e.target.value })
  }

  onTimeChange(e) {
    console.log('TIME: ', e.target.value);
    this.setState({ time: e.target.value  })
  }

  onLocationChange(e) {
    this.setState({ location: e.target.value })
  }

  onCategoryChange(e){
    this.setState({ category: e.target.value })
  }

  onImageUrlChange(e) {
    this.setState({ imageUrl: e.target.value })
  }

  onDescriptionChange(e) {
    this.setState({ description: e.target.value })
  }

  onFocusChange({ focused }) {
    this.setState(() => ({ calendarFocused: focused }))
  }

  render() {
    return (
      <div className='form'>
        <form onSubmit={this.onFormSubmit} >
          <label> Event Name: </label>
          <input
            type='text'
            onChange={this.onNameChange}
            value={this.state.name}
          />

          <label> Event Date: </label>
          <SingleDatePicker
            date={this.state.date}
            onDateChange={ date => this.setState({ date: date }) }
            focused={this.state.focused}
            onFocusChange={({ focused }) => this.setState({ focused })}
            numberOfMonths={1}
            isOutsideRange={() => false} 
          />

          <label>Time:</label>
          <input type='time' onChange={this.onTimeChange}/>
          <label> Event Location: </label>
          <input
            type='text'
            onChange={this.onLocationChange}
            value={this.state.location}
          />

          <label>Category:</label>          
          <select onChange={this.onCategoryChange} value={this.state.category ? this.state.category : ''}>
            <option disabled selected value=''>Select A Category</option>
            <option value="Sports">Sports</option>
            <option value="Entertainment">Entertainment</option>
            <option value="Food">Food</option>
            <option value="Music">Music</option>
            <option value="Miscellaneous">Miscellaneous</option>
          </select>

          <label>Image URL:</label>
          <input
            type='text'
            onChange={this.onImageUrlChange}
            value={this.state.imageUrl}
          />

          <label> Description: </label>
          <textarea
            onChange={this.onDescriptionChange}
            value={this.state.description}
          />

          <button type='submit'>Submit</button>
        </form>
      </div>
    )
  }
}

export default EventForm;