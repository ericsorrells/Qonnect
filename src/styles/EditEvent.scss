.edit-event {
	margin: 30px auto;
	display: flex;
	justify-content: center;
}

.border {
	border: 1px solid $dark-grey;
}