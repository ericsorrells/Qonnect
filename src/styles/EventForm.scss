.form {
  display: flex;
  padding-top: 10px;
  padding-right: 20px;
}

.SingleDatePicker,
.SingleDatePicker_1,
.form__element {
  width: 100%;
  margin-top: 5px;
  margin-bottom: 15px;
  border-radius: 5px;
  border: none;
  color: $dark-grey;
  background-color: white;
}

.form__select,
.form__input {
  height: 40px;
}

.form__textarea {
  height: 75px;
}

.button__purple {
  background-color: $main-color;
  width: 100%;
  color: white;
  padding: 10px 30px;
  border: none;
  border-radius: 5px;
  height: 40px;
}

/* DatePicker Styles
=========================================*/

.SingleDatePickerInput__withBorder {
  border: none;
  border-radius: 5px;
}

.DateInput_input__focused {
  border-bottom: none; // color of bottom bar when selected
}

.CalendarDay__selected_span {
  background: rgb(144, 130, 224);
  color: rgb(110, 3, 8);
}

.CalendarDay__hovered_span,
.CalendarDay__hovered_span:hover{
  background: $main-color-light;
  color: white;
}

.CalendarDay__selected,
.CalendarDay__selected:active,
.CalendarDay__selected:hover {
  background: $main-color;
  color: white;
  border: 1px solid $main-color;
}

.CalendarDay__selected_span:active,
.CalendarDay__selected_span:hover {
  border: 1px solid $dark-grey;
  border-color: 1px solid $dark-grey;
}