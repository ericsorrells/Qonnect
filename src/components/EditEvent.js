// ========================================================================================
import React from 'react';
// ========================================================================================
import EventForm from '../components/EventForm';
// ========================================================================================

// TODO: Can this be a stateless component???
class EditEvent extends React.Component {
  constructor(props) {
    super(props)
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(event) {
    const { id, eventName, date, time, location, description } = event;
    // TODO: add support for persisting edited data to Firebase
    this.props.editEvent(id, { ...event })
    this.props.history.push(`/show-event/${id}`); 
  }

  render() {
    return (
      <div>
        <EventForm
          event={this.props.event}
          onSubmit={this.onSubmit}
          profile={this.props.profile}
          auth={this.props.auth}
        />
      </div>
    )
  }
}

export default EditEvent;