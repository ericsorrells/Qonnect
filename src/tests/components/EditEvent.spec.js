// ========================================================================================
import React from 'react';
import { shallow } from 'enzyme';
import 'jest-enzyme';
// ========================================================================================
import EditEvent from '../../components/EditEvent';
import EventForm from '../../components/EventForm';
// ========================================================================================
import { basicEvent } from '../stubs/event';
import { basicProfile } from '../stubs/profile'
// ========================================================================================

describe('<EditEvent />', () => {
	describe('basic functionality', () => {
		let userId, history, onSubmit, component, match, startEditEvent;
		beforeEach(() => {
			userId = { uid: 'abc123' };
			history = { push: jest.fn() };
			onSubmit = jest.fn();
			startEditEvent = jest.fn();
			match = { params: { id: 'xyz789' } }
			component = shallow(
				<EditEvent
				event={basicEvent}
				profile={basicProfile}
				auth={userId}
				history={history}
				onSubmit={onSubmit}
				match={match}
				startEditEvent={startEditEvent}
				/>
			);
		});
		
		it('should render correctly', () => {
			expect(component).toMatchSnapshot();
		});

		it('should render all child elements', () => {
			expect(component.containsAllMatchingElements([
				<EventForm />,
			])).toBeTruthy();
		});
		
		it('should handle onSubmit', () => {
			const urlParam = match.params.id;
			component.find('EventForm').prop('onSubmit')(basicEvent);
			expect(startEditEvent).toHaveBeenLastCalledWith(urlParam, basicEvent);
		});
	});
		
	describe('secureRoute', () => {
		it('will prevent anauthorized editing: will re-route if user is not the event owner', () => {
			const userId = { uid: 'abc123' };
			const history = { push: jest.fn() };
			const onSubmit = jest.fn();
			const startEditEvent = jest.fn();
			const match = { params: { id: 'xyz789' } }
			const component = shallow(
				<EditEvent
				event={basicEvent}
				profile={basicProfile}
				auth={userId}
				history={history}
				onSubmit={onSubmit}
				match={match}
				startEditEvent={startEditEvent}
				/>
			);
			expect(history.push).not.toHaveBeenCalledWith('/not-found');
		});
		
		it('will prevent anauthorized editing: will re-route if user is not the event owner', () => {
			const userId = { uid: 'xyz789' };    // userId !== event.uid
			const history = { push: jest.fn() };
			const onSubmit = jest.fn();
			const startEditEvent = jest.fn();
			const match = { params: { id: 'xyz789' } }
			const component = shallow(
				<EditEvent
				event={basicEvent}
				profile={basicProfile}
				auth={userId}
				history={history}
				onSubmit={onSubmit}
				match={match}
				startEditEvent={startEditEvent}
				/>
			);
			expect(history.push).toHaveBeenCalledWith('/not-found');
		});
	});
});