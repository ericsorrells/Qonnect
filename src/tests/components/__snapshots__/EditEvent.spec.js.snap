// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditEvent /> basic functionality should render correctly 1`] = `
<div
  className="edit-event"
>
  <EventForm
    auth={
      Object {
        "uid": "abc123",
      }
    }
    event={
      Object {
        "category": "sports",
        "date": 345600000,
        "description": "Lorem ipsum text",
        "eventName": "Cool Event Title",
        "id": "-LDTTYL9PSqqroO6JLYh",
        "imageUrl": "https://image.com",
        "interestedUsers": "none",
        "location": "123 Elm St.",
        "selectedUser": "none",
        "time": "02:13",
        "uid": "abc123",
        "uninterestedUsers": "none",
        "userId": "abc123",
        "userName": "Rachel Deckard",
      }
    }
    onSubmit={[Function]}
    profile={
      Object {
        "acceptedEvents": Object {
          "-LC1oouVZ2uS2WVF1GNR": false,
          "-LCaSYc1PdYqDHpfxbyk": false,
          "-LCaig9Xu1qk07j-qclP": false,
        },
        "birthDate": "1976-11-04",
        "city": "The Sprawl!!!",
        "createdAt": 1526041717660,
        "description": "I am a replicant! I think. I know...!",
        "firstName": "Rachel",
        "lastName": "Deckard",
        "photoURL": "https://ih0.redbubble.net/image.14393202.3421/flat,800x800,070,f.u1.jpg",
        "state": "Ga",
        "updatedAt": 1527427279191,
        "userEvents": Object {
          "-LCfyBNNDMw6CHxm0uQT": false,
          "-LCxprm-OcVySt_LXZiz": false,
          "-LDMwtJqajA7RCImcxS1": false,
        },
      }
    }
  />
</div>
`;
